posts/Yoann/2017-06-12-hibernate-search-5-8-0-Beta3.adoc

```diff
@@ -0,0 +1,238 @@
+The reason normalizers are useful is sortable fields.
+When a field is sortable, it should never, ever be tokenized,
+because this would make sorts unpredictable:
+the sort could apply to the first token if you're lucky,
+but also to the second, or the third, etc. if you're not. 
+
+From 5.8.0.Beta3 on, Hibernate Search will log warnings whenever you're using an analyzer on a sortable field.
+If you want to get rid of the warning, just use a normalizer and everything will be fine.
+
+In Lucene, normalizers are just here to help, they work exactly as analyzers.
+The two differences are that you can't affect a tokenizer to a normalizer when defining it,
+and that normalizers have a runtime safety net: should you manage to create multiple tokens,
+Hibernate Search will concatenate them back to a single token and log a warning.
+
+In Elasticsearch 5.2+, a normalizer will be translated to a native Elasticsearch normalizer,
+and a text field with a normalizer will take the `keyword` datatype.
+
+In Elasticsearch 5.1 and below, native normalizers are not available,
+thus normalizers are simply translated to analyzers
+and a text field with a normalizer will take the `text` (5.x) or `string` (2.x) datatype.
+
+You can find out more about normalizers in the reference documentation:
+
+ * https://docs.jboss.org/hibernate/search/5.8/reference/en-US/html_single/#section-normalizers[main documentation]
+ * https://docs.jboss.org/hibernate/search/5.8/reference/en-US/html_single/#_normalizers[Elasticsearch specifics]
+
+[[es-http-client-configurer]]
+=== AWS compatibility
+
+AWS requires http://docs.aws.amazon.com/general/latest/gr/signing_aws_api_requests.html[some specific, dynamically computed headers in HTTP requests]
+to handle authentication, which until now has made it difficult to use Hibernate Search with an AWS-hosted Elasticsearch.
+
+We introduced a new SPI allowing low-level configuration of the HTTP client,
+which allows you to plug in the code required to do AWS authentication
+(and will probably also work for other cloud providers).
+
+What this means is that we currently have all of our test suite running successfully
+against a (non-public) Elasticsearch cluster on AWS,
+and will soon https://hibernate.atlassian.net/browse/HSEARCH-2772[introduce an AWS module]
+so that you can use AWS without worrying about the technical details of authentication.
+
+[[bean-resolver]]
+=== Dependency injection in FieldBridges
+
+As part of https://hibernate.atlassian.net/browse/HSEARCH-1316[HSEARCH-1316],
+we're experimenting with integration with various dependency injection frameworks.
+
+Basically, the integration is about allowing you to use annotations such as `@Inject`,
+`@PostConstruct` and so on in your FieldBridges,
+which may for instance allow you to fetch additional data
+from your application when indexing a given bean.
+
+Integration is known to work with Spring DI and CDI.
+There is no package for user consumption as of now,
```

Comment:
That's the million dollar question... Until now we didn't provide packages, though, so I suppose the natural course of action would be either wait for integrators to catch up or contribute something to Spring/Wildfly.
In any case, I suspect releasing packages in our own release train could become a massive headache (having to support a range of versions in Hibernate ORM, Spring, and WildFly at the same time...).